detectorCommon/src/xtalk_neighbors_moduleThetaMergedSegmentation.cpp

```diff
@@ -0,0 +1,224 @@
+#include "detectorCommon/DetUtils_k4geo.h"
+#include "detectorCommon/xtalk_neighbors_moduleThetaMergedSegmentation.h"
+
+// DD4hep
+#include "DDG4/Geant4Mapping.h"
+#include "DDG4/Geant4VolumeManager.h"
+
+#include <iostream>
+
+#include <unordered_set>
+
+namespace det {
+namespace crosstalk {
+
+// use it for module-theta merged readout (FCCSWGridModuleThetaMerged_k4geo)
+std::vector<std::pair<uint64_t, double>> getNeighboursModuleThetaMerged(const dd4hep::DDSegmentation::FCCSWGridModuleThetaMerged_k4geo& aSeg,
+						   const dd4hep::DDSegmentation::BitFieldCoder& aDecoder,
+						   const std::vector<std::string>& aFieldNames,
+						   const std::vector<std::vector<std::pair<int, int>>>& aFieldExtremes_layer,
+						   uint64_t aCellId) {
+
+  std::vector<std::pair<uint64_t, double>> xtalk_neighbours;
+  
+  // check that field names and extremes have the proper length
+  if (aFieldNames.size() != 3 || aFieldExtremes_layer[0].size()!=3) {
+    std::cout << "ERROR: the vectors aFieldNames and aFieldSizes should be of length = 3, corresponding to the theta/module/layer fields" << std::endl;
+    std::cout << "ERROR: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }  
+  // find index of layer, module and theta in the field name vector
+  int idModuleField(-1);
+  int idThetaField(-1);
+  int idLayerField(-1);
+  for (int itField = 0; itField < aFieldNames.size(); itField++) {
+    if (aFieldNames[itField] == aSeg.fieldNameModule())
+      idModuleField = itField;
+    else if (aFieldNames[itField] == aSeg.fieldNameTheta())
+      idThetaField = itField;
+    else if (aFieldNames[itField] == aSeg.fieldNameLayer())
+      idLayerField = itField;
+   
+  }
+  if (idModuleField < 0) {
+    std::cout << "WARNING: module field " << aSeg.fieldNameModule() << " not found in aFieldNames vector" << std::endl;
+    std::cout << "WARNING: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }
+  if (idThetaField < 0) {
+    std::cout << "WARNING: theta field " << aSeg.fieldNameTheta() << " not found in aFieldNames vector" << std::endl;
+    std::cout << "WARNING: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }
+  if (idLayerField < 0) {
+    std::cout << "WARNING: layer field " << aSeg.fieldNameLayer() << " not found in aFieldNames vector" << std::endl;
+    std::cout << "WARNING: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }
+  
+  double dummy_xtalk=0;
+  double dummy_xtalk_radial=0.7e-2; // type 1
+  double dummy_xtalk_theta=0.2e-2; // type 2
+  double dummy_xtalk_diagonal=0.04e-2; // type 3
+  double dummy_xtalk_tower=0.1e-2; // type 4
```

Comment:
```suggestion
  // crosstalk coefficients
  // see also https://indico.cern.ch/event/1368231/contributions/5904291/
  const double dummy_xtalk_radial = 0.7e-2; // type 1
  const double dummy_xtalk_theta = 0.2e-2; // type 2
  const double dummy_xtalk_diagonal = 0.04e-2; // type 3
  const double dummy_xtalk_tower = 0.1e-2; // type 4
```

Reply:
"dummy_xtalk" has been moved to the loop later in the code.
The reference to Juska's slides is added.